### ISSUE TITLE
[Visual Bug] Optifine affects shading of slope blocks

### ISSUE THREAD

**Issue by Dare0900** [sp614x/optifine#5673]
## Description of Issue
Optifine seems to cause visual glitches with modded slope blocks that don't occur without optifine. Tested with both the "Colorizer Slope" block from [AssortedDecor](https://www.curseforge.com/minecraft/mc-mods/assorted-decor) and the "Frame Slope" block from [Blockcarpentry](https://www.curseforge.com/minecraft/mc-mods/blockcarpentry). Any time there is a solid block near these slopes their shadows become distorted and result in one side of the block being far darker than the other. The lighting varies a lot between faces. 

In the images below, the "Frame Slope" blocks are the top row, and the "Colorizer Slope" blocks are the middle row. The bottom row has 2 slopes from each mod not touching any solid blocks as a control. 

![image](https://user-images.githubusercontent.com/60798121/119537390-3980c480-bd47-11eb-918e-cbd85dd3da77.png)
Without Optifine, lighting is not 100% perfect but shadows generally seem to go where they should

![image](https://user-images.githubusercontent.com/60798121/119537877-bd3ab100-bd47-11eb-92b5-fb424f60d79f.png)
With Optifine, smooth lighting OFF

![image](https://user-images.githubusercontent.com/60798121/119538897-cbd59800-bd48-11eb-935c-a12552e9dea2.png)
With Optifine, smooth lighting Maximum + 100%

## Steps to Reproduce
Install any mod that introduced customizable slope blocks, install optifine

## OptiFine Version
1.16.5 HD U G7 is what I was testing when I took the screenshots, but the same issue occurs with G8

## Installation Method
.jar was placed in mods folder

##Forge Version
Tested with Forge 36.1.0 and up, issue present in all versions

## Other Installed Mods
Forge 36.1.0, AssortedDecor 1.16.5-2.0.13, Blockcarpentry 1.16-0.3.5a
Tested with each slope mod installed individually, the same issues occur. 

## F3 Debug Screenshot
![image](https://user-images.githubusercontent.com/60798121/119536759-aa73ac80-bd46-11eb-8b72-d8e61276799e.png)

## Does this happen without OptiFine?
- No

**Comment by KaiAF** [comment]
Does this all happen with internal shaders disabled? I would assume it would still happen, but just asking

**Comment by Dare0900** [comment]
Apologies for the late reply - 
This still occurs with Antisitropic filtering disabled, and that actually causes further lighting glitches with slope blocks from yet another mod, FramedBlocks
The issue does persist with no internal shaders, though it changes the shading from what you see in the images above with one corner of a face being very dark and the other light to making the entire face dark if its near a solid block
Let me know if you want more details and/or screenshots

**Comment by KriZ04** [comment]
Also happening in the blockcraftery mod with _Optifine 1.12.2 HD U G5_.

**Comment by MrRedstoneA1** [comment]
yep, this is still happening with LittleTiles slopes as well as blockcraftery. Even if you edit a vanilla block and make a custom json model with a 45 degree wall, it gets affected. The only "fix" I found is making sure the slopes are always at light level 15.
